src/BindPattern.jl

```diff
@@ -122,12 +124,15 @@ function bind_pattern!(
     elseif @capture(source, T_(subpatterns__)) && is_possible_type_name(T)
         # struct pattern.
         len = length(subpatterns)
-        named_fields = [pat.args[1] for pat in subpatterns if (pat isa Expr) && pat.head == :kw]
+        named_fields = [pat.args[1] for pat in subpatterns
+                                    if (pat isa Expr) && pat.head == :kw]
```

Comment:
would be nice to switch this over to the `is_expr(pat, :kw)` syntax, but would need to import that method on this branch as well

Reply:
Part of the reason I didn't do that is that the function doesn't exist in earlier Julia versions.

Reply:
ah, does the other branch add a static if on version to backport the method?